### ISSUE TITLE
Memory usage report incorrect

### ISSUE THREAD

**Issue by gambiteer** [gambit/gambit#201]
I'm on Raspberry Pi 2 B:

cutie-pi:~/programs/gambit> gsi -v
v4.8.5 20160608210716 armv7l-unknown-linux-gnueabihf "./configure 'CC=gcc -mcpu=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard' '--enable-single-host' '--enable-shared' '--enable-multiple-versions'"

and "time" seems to vastly under-report memory usage, e.g.,

```
> (define a (time (expt 3 1000000)))(define b (time (* a a)))
(time (expt 3 1000000))
    526 ms real time
    520 ms cpu time (470 user, 50 system)
    3 collections accounting for 11 ms real time (0 user, 0 system)
    477232 bytes allocated
    1766 minor faults
    no major faults
> (time (* a a))
    534 ms real time
    530 ms cpu time (510 user, 20 system)
    1 collection accounting for 3 ms real time (0 user, 0 system)
    32552 bytes allocated
    2154 minor faults
    no major faults
> (/ (integer-length b) 8.)
396240.75
```

So for the last computation it doesn't report enough bytes even to hold the result.

**Comment by gambiteer** [comment]
Marc:  Even after your fix for the overflow problem in computing heap size, this error remains.

**Comment by adam-7** [comment]
@gambiteer , wait can you clarify the problem here? i'm not following exactly.:

on amd64 i get 9MB allocations for the expt. and 8MB for the (* a a).

what you presented above is a garbled console interaction, right, first you do (define a ..) (define b ..) and then their evaluations are reprinted below - those are intended to represent in total two evaluations and not four evaluations as could be understood from your text here, right?

also what's your point with the (/ .. 8.)?

**Comment by gambiteer** [comment]
1.  The computations are on arm32, a raspberry pi.
2.  I don't think the interaction is "garbled", I just put two statements on one line at the top.
3.  I wanted to know how many bytes were occupied by b, so I divided the number of bits by 8.

**Comment by 0-8-15** [comment]
Garbage collector reports seem to be affected as well.  Note that we have a negative total of allocations (after only about a minute of runtime and this being the 4th GC since startup):

    *** GC: 73ms, -11328.9K alloc, 9.6M heap, 2.7M live (28% 2233760+637520)

**Comment by gambiteer** [comment]
I believe this is fixed with e529b1d6d5b7bae2fcc3e9cd8e7f3e11a1318193.
